ui/src/app/panel/thread-list/thread-list.component.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {Observable} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+import {DataService} from '../../service/data.service';
+
+@Component({
+  templateUrl: './thread-list.component.html',
+  styleUrls: ['./thread-list.component.sass']
+})
+export class ThreadListComponent implements OnInit {
+  threadList$: Observable<any>;
+  selectedId: number;
+
+  constructor(private dataService: DataService,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.threadList$ = this.route.paramMap.pipe(
+      switchMap(params => {
+        this.selectedId = +params.get('id');
+        return this.dataService.threadList;
```

Comment:
This looks like having the same effect as `this.threadList$ = this.dataService.threadList`.